prow/cmd/deck/static/prow/prow.ts

```diff
@@ -674,6 +704,127 @@ function stateToAdj(state: JobState): string {
     }
 }
 
+function parseDuration(duration: string): number {
+    if (duration.length === 0) {
+        return 0;
+    }
+    let seconds = 0;
+    let multiple = 0;
+    for (let i: number = duration.length; i >= 0; i--) {
+        const ch = duration[i];
+        if (ch === 's') {
+            multiple = 1;
+        } else if (ch === 'm') {
+            multiple = 60;
+        } else if (ch === 'h') {
+            multiple = 60 * 60;
+        } else if (ch >= '0' && ch <= '9') {
+            seconds += Number(ch) * multiple;
+            multiple *= 10;
+        }
+    }
+    return seconds;
+}
+
+function formatDuration(seconds: number): string {
+    const parts: string[] = [];
+    if (seconds > 3600) {
+        const hours = Math.floor(seconds / 3600);
+        parts.push(String(hours));
+        parts.push('h');
+        seconds = seconds % 3600;
+    }
+    if (seconds > 60) {
+        const minutes = Math.floor(seconds / 60);
+        if (minutes > 0) {
+            parts.push(String(minutes));
+            parts.push('m');
+            seconds = seconds % 60;
+        }
+    }
+    if (seconds > 0) {
+        parts.push(String(seconds));
+        parts.push('s');
+    }
+    return parts.join('');
+}
+
+function drawJobHistogram(total: number, jobHistogram: JobHistogram, start: number, end: number): void {
+    const startEl = document.getElementById("job-histogram-start") as HTMLSpanElement;
+    if (startEl != null) {
+        startEl.textContent = `${formatDuration(end - start)} ago`;
+    }
+
+    // make sure the empty table is hidden
+    const tableEl = document.getElementById("job-histogram") as HTMLTableElement;
+    const labelsEl = document.getElementById("job-histogram-labels") as HTMLDivElement;
+    if (jobHistogram.length() === 0) {
+        tableEl.style.display = "none";
+        labelsEl.style.display = "none";
+        return;
+    }
+    tableEl.style.display = "";
+    labelsEl.style.display = "";
+
+    const el = document.getElementById("job-histogram-content") as HTMLTableSectionElement;
+    el.title = `Showing ${jobHistogram.length()} builds from last ${formatDuration(end - start)} by start time and duration, newest to oldest.`;
+    const rows = 10;
+    const width = 12;
+    const cols = Math.round(el.clientWidth / width);
+
+    // initialize the table if the row count changes
+    if (el.childNodes.length !== rows) {
+        el.innerHTML = "";
+        for (let i: number = 0; i < rows; i++) {
+            const tr = document.createElement('tr');
+            for (let j: number = 0; j < cols; j++) {
```

Comment:
`number` is inferred and can be omitted.

```suggestion
            for (let j = 0; j < cols; j++) {
```